### PR TITLE
Tests for batch job monitoring and management.

### PR DESCRIPTION
**Pull request by zebrik**

### CODE REVIEW

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    public BatchJobOperations(final OnlineManagementClient client) {
+        this.client = client;
+        this.ops = new Operations(client);
+    }
+
+    public long startJob(final String deploymentName, final String jobXmlName) throws IOException {
+        ModelNodeResult result = ops.invoke("start-job", getBatchSubsystemInDeploymentAddress(deploymentName),
+                Values.of("job-xml-name", jobXmlName));
+        result.assertDefinedValue();
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java:44]: Shouldn't be here `assertSuccess()` ?

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        HOST_NAME = ConfigUtils.getDefaultHost();
+    private final OnlineManagementClient client;
+    private final Operations ops;
+
+    public BatchJobOperations(final OnlineManagementClient client) {
+        this.client = client;
+        this.ops = new Operations(client);
+    }
+
+    public long startJob(final String deploymentName, final String jobXmlName) throws IOException {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java:41]: Provide javadoc, mainly explain the return value.

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchManagementTestCase.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+            assertEquals(expectedAttrValueFromModel, actualAttrValueFromBrowser);
+        }
+    }
+
+    private boolean everyJobFileNameInTableEquals(String expectedJobFileName) {
+        return page.getAllRows().stream().map(row -> page.getJobFileNameFromRow(row))
+                .allMatch(jobFileName -> jobFileName.equals(expectedJobFileName));
+    }
+
+    private void deployBatchApplication(long timeout) throws CommandFailedException {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchManagementTestCase.java]: As heavily used method in tests, please provide javadoc. It is good to explain, how many batch jobs it contains.

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/Processor.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import javax.batch.api.BatchProperty;
+import javax.batch.api.chunk.ItemProcessor;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named("testProcessor")
+public class Processor implements ItemProcessor {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/Processor.java:38]: Javadoc please.

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/Reader.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.util.logging.Logger;
+
+import javax.batch.api.chunk.ItemReader;
+import javax.inject.Named;
+
+@Named("testReader")
+public class Reader implements ItemReader {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/Reader.java:38]: Javadoc please

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/Writer.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.batch.api.chunk.AbstractItemWriter;
+import javax.inject.Named;
+
+@Named("testWriter")
+public class Writer extends AbstractItemWriter {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/Writer.java:37]: Javadoc please

**Comment by rhatlapa** [comment]
Generally LGTM, please just update the javadocs.

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java (hunk 1 above)
Comment (zebrik) [reply basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java:44]: `assertSuccess()` is part of `assertDefinedValue()` and we are interested in the value containing the execution id as well.

Comment (zebrik) [reply basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java:44]: I will add javadoc about the returned value.

Comment (rhatlapa) [reply basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchJobOperations.java:44]: you are right about the `assertDefinedValue()` and `assertSuccess()`, I forgot it is incorporated.

**Comment by zebrik** [comment]
@rhatlapa please review again.

**Comment by rhatlapa** [comment]
Thanks, LGTM, merging

**Pull request merged**